### PR TITLE
feat:Support multi-discovery server.

### PR DESCRIPTION
**Pull request by SkyeBeFreeman**

### CODE REVIEW

**Comment by SkyeBeFreeman** [comment]
implement #59

File: polaris-plugins/polaris-plugins-connector/connector-composite/src/main/java/com/tencent/polaris/plugins/connector/composite/CompositeServiceUpdateTask.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                        boolean needAdd = true;
+                        for (Instance j : polarisInstanceList) {
+                            if (i.getHost().equals(j.getHost().getValue()) && i.getPort() == j.getPort().getValue()) {
+                                needAdd = false;
+                                break;
+                            }
+                        }
+                        if (needAdd) {
+                            Instance instance = Instance.newBuilder()
+                                    .setNamespace(StringValue.of("default"))
```
Comment (lepdou) [inline polaris-plugins/polaris-plugins-connector/connector-composite/src/main/java/com/tencent/polaris/plugins/connector/composite/CompositeServiceUpdateTask.java:110]: namespace will be better as same as polaris instance's namespace?

File: polaris-plugins/polaris-plugins-connector/connector-composite/src/main/java/com/tencent/polaris/plugins/connector/composite/CompositeServiceUpdateTask.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                DiscoverResponse discoverResponse = (DiscoverResponse) serverEvent.getValue();
+                DiscoverResponse.Builder newDiscoverResponseBuilder = DiscoverResponse.newBuilder()
+                        .mergeFrom(discoverResponse);
+                CompositeConnector connector = (CompositeConnector) serverConnector;
+                if (EventType.INSTANCE.equals(serviceEventKey.getEventType())) {
+                    // Get instance information list except polaris.
+                    List<DefaultInstance> extendInstanceList = new ArrayList<>();
+                    for (DestroyableServerConnector sc : connector.getServerConnectors()) {
+                        if (!DefaultPlugins.SERVER_CONNECTOR_GRPC.equals(sc.getName())) {
+                            List<DefaultInstance> instanceList = sc.syncGetServiceInstances(this);
```
Comment (lepdou) [inline polaris-plugins/polaris-plugins-connector/connector-composite/src/main/java/com/tencent/polaris/plugins/connector/composite/CompositeServiceUpdateTask.java:90]: should handle exception for sc.syncGetServiceInstances(this)?

File: polaris-plugins/polaris-plugins-connector/connector-consul-api/src/main/java/com/tencent/polaris/plugins/connector/consul/ConsulAPIConnector.java [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+            CommonProviderResponse resp = new CommonProviderResponse();
+            id = service.getId();
+            resp.setInstanceID(service.getId());
+            resp.setExists(true);
+            LOG.info("Registered service to Consul: " + service);
+            return resp;
+        } catch (ConsulException e) {
+            LOG.warn("Register instance to Consul failed of service: " + service, e);
+        }
+        return null;
```
Comment (lepdou) [inline polaris-plugins/polaris-plugins-connector/connector-consul-api/src/main/java/com/tencent/polaris/plugins/connector/consul/ConsulAPIConnector.java:147]: shoud wrap exception to PolarisException?  can refer to GrpcConnector .

And shoud not return null

File: polaris-plugins/polaris-plugins-connector/connector-consul-api/src/main/java/com/tencent/polaris/plugins/connector/consul/ConsulAPIConnector.java [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+            service.setCheck(check);
+        }
+        return service;
+    }
+
+    @Override
+    public void deregisterInstance(CommonProviderRequest req) throws PolarisException {
+        LOG.info("Unregistering service to Consul: " + id);
+        this.consulClient.agentServiceDeregister(id);
+        LOG.info("Unregistered service to Consul: " + id);
```
Comment (lepdou) [inline polaris-plugins/polaris-plugins-connector/connector-consul-api/src/main/java/com/tencent/polaris/plugins/connector/consul/ConsulAPIConnector.java:174]: should handle exception

File: polaris-plugins/polaris-plugins-connector/connector-consul-api/src/main/java/com/tencent/polaris/plugins/connector/consul/ConsulAPIConnector.java [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+    }
+
+    @Override
+    public void heartbeat(CommonProviderRequest req) throws PolarisException {
+        this.consulClient.agentCheckPass("service:" + id);
```
Comment (lepdou) [inline polaris-plugins/polaris-plugins-connector/connector-consul-api/src/main/java/com/tencent/polaris/plugins/connector/consul/ConsulAPIConnector.java:179]: should handler exception

File: polaris-plugins/polaris-plugins-connector/connector-consul-api/src/main/java/com/tencent/polaris/plugins/connector/consul/ConsulAPIConnector.java [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+        }
+    }
+
+    @Override
+    public List<DefaultInstance> syncGetServiceInstances(ServiceUpdateTask serviceUpdateTask) {
+        HealthServicesRequest request = HealthServicesRequest.newBuilder()
+                .setQueryParams(new QueryParams(ConsistencyMode.DEFAULT))
+                .build();
+        Response<List<HealthService>> response = this.consulClient
+                .getHealthServices(serviceUpdateTask.getServiceEventKey().getService(), request);
```
Comment (lepdou) [inline polaris-plugins/polaris-plugins-connector/connector-consul-api/src/main/java/com/tencent/polaris/plugins/connector/consul/ConsulAPIConnector.java:191]: should handler exception

File: polaris-plugins/polaris-plugins-connector/connector-consul-api/src/main/java/com/tencent/polaris/plugins/connector/consul/ConsulAPIConnector.java [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+        }
+        return instanceList;
+    }
+
+    @Override
+    public Services syncGetServices(ServiceUpdateTask serviceUpdateTask) {
+        CatalogServicesRequest request = CatalogServicesRequest.newBuilder()
+                .setQueryParams(QueryParams.DEFAULT).build();
+        ArrayList<String> serviceList = new ArrayList<>(
+                this.consulClient.getCatalogServices(request).getValue().keySet());
```
Comment (lepdou) [inline polaris-plugins/polaris-plugins-connector/connector-consul-api/src/main/java/com/tencent/polaris/plugins/connector/consul/ConsulAPIConnector.java:211]: should handler exception

File: polaris-common/polaris-config/src/main/java/com/tencent/polaris/factory/config/global/ServerConnectorConfigImpl.java [hunk 8]
```diff
@@ -142,6 +136,11 @@ public void setReconnectInterval(Long reconnectInterval) {
         this.reconnectInterval = reconnectInterval;
     }
 
+    @Override
+    public Map<String, String> getMetadata() {
```
Comment (andrewshan) [inline polaris-common/polaris-config/src/main/java/com/tencent/polaris/factory/config/global/ServerConnectorConfigImpl.java:140]: 这个metadata的用处是啥？

Comment (SkyeBeFreeman) [reply polaris-common/polaris-config/src/main/java/com/tencent/polaris/factory/config/global/ServerConnectorConfigImpl.java:141]: consul会有一些额外的配置，放在这里

**Pull request merged**